### PR TITLE
Hyperparameter tuning

### PR DESCRIPTION
**Pull request by jkarolczak**

### CODE REVIEW

**Comment by jkarolczak** [comment]
meme

File: cfg/tune_hparams.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+train_cfg: "../cfg/train.yaml"
+epochs: 2
```
Comment (wtaisner) [inline cfg/tune_hparams.yaml:2]: shouldn't this value be well, higher?

File: cfg/tune_hparams.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+n_calls: 10
+hparams:
```
Comment (wtaisner) [inline cfg/tune_hparams.yaml:4]: perhaps we can also tune gradient accumulation?

File: scripts/tune_hparams.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
+import gc
+import os
+import random
+from datetime import datetime
+from typing import Any, Dict, List, Tuple
+
+import torch
+import numpy as np
+import MinkowskiEngine as ME
+import skopt
```
Comment (wtaisner) [inline scripts/tune_hparams.py:10]: AS is proud of you, my dude

File: scripts/tune_hparams.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            log.epoch(run=run, preds=predictions, target=groundtruth, epoch_num=e)
+
+        score = run["eval/accuracy"].fetch_values().max().value
+
+        run.stop()
+        return score
+
+
+    dims = construct_dims(cfg["hparams"])
+    best_hparams = skopt.gp_minimize(evaluate, dims, n_calls=cfg["n_calls"]).x
```
Comment (wtaisner) [inline scripts/tune_hparams.py:156]: PTSD kick in

**Review by wtaisner — COMMENTED** [review]
let's discussed raised issues: my PTSD and other hyperparameters' tuning

**Comment by wtaisner** [comment]
image

i fixed your meme btw

File: cfg/tune_hparams.yaml (hunk 2 above)
Comment (jkarolczak) [reply cfg/tune_hparams.yaml:4]: We can. However this may cause an issue with type. I will try to wrap proper snippet with int(...) to prevent the issue. I proposed hparams only as an example, showoff how to use new functionality

Comment (jkarolczak) [reply cfg/tune_hparams.yaml:4]: It can be specified as:
```yaml
accum_iter:
    space: integer
    lower_bound: 1
    upper_bound: 16
```
and voilà!

**Review by wtaisner — APPROVED** [review]
AS wants to hire you
